### PR TITLE
Bump to xamarin/Java.Interop/main@3824b974

### PR DESCRIPTION
**Pull request by jonpryor**
Fixes: https://github.com/xamarin/java.interop/issues/790

Changes: https://github.com/xamarin/java.interop/compare/bba1f077ae9886cdcc3e35fee35386937ce885dc...3824b974dada6f302ed5f7b7f97ffbd08990a82a

  * xamarin/java.interop@3824b974: [java-interop] Windows build system support (#816)
  * xamarin/java.interop@94c0c709: Bump to xamarin/xamarin-android-tools/main@554d45a (#813)
  * xamarin/java.interop@5c756b14: [Java.Interop-PerformanceTests] Support .NET Core 3.1 (#808)
  * xamarin/java.interop@daec07b6: [build] Fix various warnings (#812)
  * xamarin/java.interop@678c4bd2: [class-parse, generator] Allow showing Kotlin internals via metadata (#793)
  * xamarin/java.interop@cd4c8f80: [jnienv-gen] Generate a header file for the native functions (#809)
  * xamarin/java.interop@69767c1a: [param-name-importer] Fix NSE when updating JavaApi.AllPackages (#807)
  * xamarin/java.interop@a666a6f9: [Java.Runtime.Environment] Partial support for .NET Core (#804)

### CODE REVIEW

**Review by grendello — APPROVED** [review]

**Comment by jonpryor** [comment]
All unit tests are green.

Windows doesn't build.  Looks like `vcvarsall.bat` might not exist? https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4570166&view=logs&j=cac0e8d3-0ef5-5d2b-b57e-e8fde7204df3&t=d0eaacf3-179f-5217-7519-cd03bc45e5e4

```
  call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat"  x86_amd64 && cmake -G "NMake Makefiles" -S "C:/a/1/s/external/Java.Interop/src/java-interop/" -B "C:/a/1/s/external/Java.Interop/src/java-interop/obj//Release/win-x64/" "-DJDK_INCLUDE_LIST=C:/Users/AzDevOps/android-toolchain/jdk-11/include;C:/Users/AzDevOps/android-toolchain/jdk-11/include/win32" "-DJNI_C_PATH=C:/a/1/s/external/Java.Interop/src/java-interop/obj//Release/jni.c" && cmake --build "C:/a/1/s/external/Java.Interop/src/java-interop/obj//Release/win-x64/" -v
  The system cannot find the path specified.
```

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.targets [hunk 1]
```diff
@@ -218,6 +218,14 @@
         DestinationFile="Xamarin.Android.Common.props"
         Replacements="@PACKAGE_VERSION@=$(ProductVersion);@PACKAGE_VERSION_BUILD@=$(XAVersionCommitCount);@NDK_ARMEABI_V7_API@=$(AndroidNdkApiLevel_ArmV7a);@NDK_ARM64_V8A_API@=$(AndroidNdkApiLevel_ArmV8a);@NDK_X86_API@=$(AndroidNdkApiLevel_X86);@NDK_X86_64_API@=$(AndroidNdkApiLevel_X86_64);@BUNDLETOOL_VERSION@=$(XABundleToolVersion)">
     </ReplaceFileContents>
+    <PropertyGroup>
+      <_HaveVcVars Condition=" Exists('C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat') ">True</_HaveVcVars>
+    </PropertyGroup>
+    <Message Text="# jonp: Does vcvarsall.bat exist? $(_HaveVcVars) " />
```
Comment (dellis1972) [inline src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.targets:224]: Do we need this message still?

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.targets [hunk 2]
```diff
@@ -218,6 +218,14 @@
...
+    <ItemGroup Condition=" '$(_HaveVcVars)' == '' And '$(OS)' == 'Windows_NT' ">
+      <_Vcvars Include="C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\**\vcvarsall.bat" />
+    </ItemGroup>
+    <Message Text="# jonp: Anywhere? @(_Vcvars, ';')" />
```
Comment (dellis1972) [inline src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.targets:228]: same here.

**Comment by jonpryor** [comment]
Welp, this is disturbing:

```
Task "Message" (TaskId:1532)
  Task Parameter:Text=# jonp: Does vcvarsall.bat exist?   (TaskId:1532)
  # jonp: Does vcvarsall.bat exist?   (TaskId:1532)
Done executing task "Message". (TaskId:1532)
Task "Message" (TaskId:1533)
  Task Parameter:Text=# jonp: Anywhere?  (TaskId:1533)
  # jonp: Anywhere?  (TaskId:1533)
```

Apparently there is no `vcvarsall.bat` on this machine.

Does it have *any* VC files?

**Comment by jonpryor** [comment]
Uh oh:

```
  # jonp: Does VC exist? True  (TaskId:1534)
  # jonp: Any scripts?
   (TaskId:1536)
```

The VC directory exists, but there's no `.bat` or `.cmd` files within it!

**Pull request merged**